.github/DeliverToBCFeed.ps1

```diff
@@ -0,0 +1,135 @@
+Param(
+    [Hashtable] $parameters
+)
+
+function GenerateManifest
+(
+    [Parameter(Mandatory=$true)]
+    $PackageId,
+    [Parameter(Mandatory=$true)]
+    $Version,
+    [Parameter(Mandatory=$true)]
+    $Authors,
+    [Parameter(Mandatory=$true)]
+    $Owners
+)
+{
+    [xml] $template = Get-Content "$PSScriptRoot\ALAppExtensions.nuspec"
+
+    $template.package.metadata.id = $PackageId
+    $template.package.metadata.version = $Version
+    $template.package.metadata.authors = $Authors
+    $template.package.metadata.owners = $Owners
+
+    return $template
+}
+
+try 
+{
+    $deliverContext = $parameters.Context | ConvertFrom-Json | ConvertTo-HashTable
+    $deliverServerUrl = $deliverContext.ServerUrl
+    $deliverAcountToken = $deliverContext.Token
+}
+catch 
+{
+    throw "Deliver context is malformed. Needs to be formatted as JSON, containing serverUrl and token."
+}
+
+if (-not ($deliverServerUrl)) 
+{
+    throw "Cannot retrieve server URL from the deliver context"
+} 
+
+if (-not $deliverAcountToken) 
+{
+    throw "Cannot retrieve account token the deliver context"
+} 
+
+Write-Host "Successfully retrieved information from the deliver context" -ForegroundColor Green
+
+$project = $parameters.project
+$projectName = $parameters.projectName
+$appsFolder = $parameters.appsFolder
+$testAppsFolder = $parameters.testAppsFolder
+$type = $parameters.type
+
+# Construct package ID
+$packageId = "$($env:GITHUB_REPOSITORY_OWNER)-$($env:RepoName)"
+
+if (-not $project -or ($project -ne '.')) 
+{
+    $packageId += "-$projectName"
+}
+
+if ($type -eq 'CD') 
+{
+    $packageId += "-preview"
+}
+
+# Extract version from the published folders (naming convention)
+$packageVersion = $appsFolder.SubString($appsFolder.LastIndexOf("-Apps-") + "-Apps-".Length) #version is right after '-Apps-'
```

Comment:
It might be a matter of taste, but I find regexes more readable than this.
This would be
``` $packageVersion = $appsFolder -replace '.*-Apps-','' #version is ... ```

Reply:
For me, it is also clearer using the regex :) Changed it. Thanks.